src/handle/command.ml

```diff
@@ -376,7 +380,24 @@ let handle : (Path.t -> Sign.t) -> sig_state -> p_command ->
             in
             (Some(a), Scope.get_implicitness a)
       in
-      let ao = Option.map scope_p ao in
+      (* Show warnings if it's not a definition. *)
+      let ao = Option.map (scope_p (p_sym_trm = None)) ao in
+      (* In case of a definition, warn if variables are named but not used
+         inside the type or the definition. *)
+      let rec binders_warn ty te =
+        match ty, te with
+        | Prod(_, by), Abst(_, be) ->
+            let x, by, be = Bindlib.unbind2 by be in
+            if not (Bindlib.occur x (lift by))
+            && not (Bindlib.occur x (lift be)) then
```

Comment:
No lifting is necessary!
```
let x, ty, te = Bindlib.unbind2 by be in
if not (Bindlib.occur x by) && not (Bindlib.occur x be) then
  wrn pos "Variable [%a] could be replaced by [_]." pp_var x;
binders_warn ty te
```

Reply:
Remember to always print identifier by using pp_var or pp_sym in order to handle escaped identifiers properly. Never print variable names using Bindlib.name_of.

Reply:
It's not well typed, `Bindlib.occur : 'a var -> 'a box`!

Reply:
Ah but `binder_constant` can be used instead, don't you think?

Reply:
Right, sorry. I was thinking in binder_occur or binder_constant indeed.